File: S5/PDS/pds/c Rappels de C/modifie.py
def modifie(n, t):
    n += 1
    t[0] += 1

def main():
    n = 0
    t = [0]

    print('avant')
    print('n: {}'.format(n))
    print('t: {}'.format(t))

    modifie(n, t)

    print('après')
    print('n: {}'.format(n))
    print('t: {}'.format(t))

main()

File: S5/PDS/pds/c Rappels de C/modifie.c
#include <stdio.h>

void modifie(int *n, int *t) {
    n[0]++;
    t[0]++;
}

int main() {
    int n = 0;
    int t[1] = { 0 };

    printf("avant\n");
    printf("n: %d\n", n);
    printf("t: [%d]\n", t[0]);

    modifie(&n, t);

    printf("après\n");
    printf("n: %d\n", n);
    printf("t: [%d]\n", t[0]);

    return 0;
}
